### PR TITLE
Check added for invalid stack analysis request id

### PR DESCRIPTION
**Pull request by samuzzal-choudhury**
Fixes issues: 

https://github.com/openshiftio/openshift.io/issues/276
https://github.com/fabric8-analytics/fabric8-analytics-server/issues/61
https://github.com/fabric8-analytics/fabric8-analytics-server/issues/22

### CODE REVIEW

File: bayesian/api_v1.py [hunk 1]
```diff
@@ -281,6 +281,9 @@ class StackAnalysesByGraphGET(ResourceWithSchema):
 
     @staticmethod
     def get(external_request_id):
+        if (get_request_count(rdb, external_request_id) < 1):
```
Comment (fridex) [inline bayesian/api_v1.py]: Brackets are not needed here.

File: bayesian/utils.py [hunk 2]
```diff
@@ -358,3 +358,10 @@ def retrieve_worker_result (rdb, external_request_id, worker):
         result = row.to_dict()
     return result
 
+def get_request_count(rdb, external_request_id):
+    try:
+        count = rdb.session.query(StackAnalysisRequest).filter(\
+                    StackAnalysisRequest.id == external_request_id).count()
+        return count
+    except SQLAlchemyError:
```
Comment (fridex) [inline bayesian/utils.py]: I think this try-except block is not needed here as we would like to get these kind of errors in logs.

File: bayesian/api_v1.py (hunk 1 above)
Comment (samuzzal-choudhury) [inline bayesian/api_v1.py]: @fridex I don't know why the C programmer in me catches me unawares at times :-(. Changed. Next commit en route.

File: bayesian/utils.py (hunk 2 above)
Comment (samuzzal-choudhury) [inline bayesian/utils.py]: Got that. Changed. Coming the in the next commit.

**Pull request merged**